frame/system/src/lib.rs

```diff
@@ -1203,12 +1231,22 @@ impl<T: Trait + Send + Sync> CheckWeight<T> where
 		let current_weight = Module::<T>::all_extrinsics_weight();
 		let maximum_weight = T::MaximumBlockWeight::get();
 		let limit = Self::get_dispatch_limit_ratio(info.class) * maximum_weight;
-		let added_weight = info.weight.min(limit);
-		let next_weight = current_weight.saturating_add(added_weight);
-		if next_weight > limit && info.class != DispatchClass::Mandatory {
-			Err(InvalidTransaction::ExhaustsResources.into())
-		} else {
+		if info.class == DispatchClass::Mandatory {
+			// If we have a dispatch that must be included in the block, we will do everything to
+			// make sure we do not return an error here.
```

Comment:
```suggestion
			// If we have a dispatch that must be included in the block, it ignores all the limits. 
```

I feel like your are sugar coding the harsh reality here :D